urls.py

```diff
@@ -16,3 +16,18 @@
         name=SolutionsPublicView.urlname),
     url(r'^supply/$', RedirectView.as_view(url='/solutions/#supply', permanent=True)),
 )
+
+urlpatterns = root_patterns + patterns(
```

Comment:
Django does provide a [prefixed url](https://docs.djangoproject.com/en/1.9/topics/i18n/translation/#language-prefix-in-url-patterns) facility, but it is limited in more than one way. It doesn't support adding `lang` prefix to create URLs like `lang/fra/home` and it doesn't allow pattern includes, it works only at root level restricting us to add `lang` prefix ourselves. There is one more limitation that is solved by a 3rd party middleware https://github.com/st4lk/django-solid-i18n-urls but it doesn't still support `lang` prefix. 

So the options are to write our own middleware that does all of what we want or be okay with this DRY breaker (repeating all URL patterns with different regex for prefix). I think this is an okay compromise since the lang prefixes are needed only for prelogin urls which won't be more than a dozen.

What do you think @czue (pinging you since are aware of the spec)